### PR TITLE
Secret dir materialized in alloc/task directory

### PR DESCRIPTION
**Pull request by dadgar**

### CODE REVIEW

**Comment by diptanu** [comment]
LGTM

File: client/allocdir/alloc_dir_linux.go [hunk 1]
```diff
@@ -23,6 +30,36 @@ func (d *AllocDir) unmountSharedDir(dir string) error {
...
+	}
+
+	return os.MkdirAll(dir, 0777)
+}
+
+// createSecretDir removes the secrets dir folder
+func (d *AllocDir) removeSecretDir(dir string) error {
+	if unix.Geteuid() == 0 {
+		if err := syscall.Unmount(dir, 0); err != nil {
+			return err
```
Comment (schmichael) [inline client/allocdir/alloc_dir_linux.go]: NewSyscallError

File: client/driver/env/env.go [hunk 2]
```diff
@@ -249,6 +257,16 @@ func (t *TaskEnvironment) ClearTaskLocalDir() *TaskEnvironment {
 	return t
 }
 
+func (t *TaskEnvironment) SetSecretDir(dir string) *TaskEnvironment {
+	t.SecretDir = dir
+	return t
+}
+
+func (t *TaskEnvironment) ClearSecretDir() *TaskEnvironment {
```
Comment (schmichael) [inline client/driver/env/env.go]: Why have setters for public fields?

**Comment by schmichael** [comment]
LGTM

File: client/driver/env/env.go (hunk 2 above)
Comment (dadgar) [inline client/driver/env/env.go]: Some do more https://github.com/hashicorp/nomad/blob/4bae54741e71ac281ea591731b384ea3719f9f2b/client/driver/env/env.go#L362

**Pull request merged**